### PR TITLE
H3: Gesten-Ursprung bei Ringpuffer-Overflow re-verankern

### PR DESCRIPTION
**Pull request by cl445**
Behebt **H3** aus dem Code-Review (`CODE_REVIEW.md`).

## Problem
Alle Touch-Punkte sind Translationen relativ zum Touch-Down; der Ursprung `(0,0)` wird als erstes Sample angehängt. Der Positions-Ringpuffer fasst `positionBufferSize = 60` Punkte. Bei langen/langsamen Gesten (>60 Samples — bei 120 Hz ProMotion in ~0,5 s erreicht) überschreibt der Puffer genau den Ursprung. Da alle **start-relativen** Features (`maxDisplacement`, `returnRatio`, dominanter Winkel, Circularity) ab `points.first` messen, wird die Geste dann gegen einen Mittelpunkt gemessen und Return-/Kreisgesten werden fehlklassifiziert.

## Fix
`KeyGestureRecognizer.anchoringOrigin(_:)` stellt `(0,0)` wieder voran, wenn `elements[0] != .zero` — was genau dem Overflow-Fall entspricht (ohne Overflow ist das erste Sample immer der angehängte Ursprung). Reine, testbare Funktion; minimaler Eingriff, Puffergröße unverändert (Memory-Budget der Extension).

## Tests
- Pure-Cases (leer / führender Ursprung / evicted).
- End-to-End: eine lange Right-Return-Swipe, deren Ursprung der Puffer verworfen hat, wird nach Re-Verankerung wieder korrekt als Return-Swipe nach rechts erkannt.
- **585 Tests grün** (iPhone 17 Pro).

## Hinweis
Basis `refactor/pr16-directory-structure`. Unabhängig von #172/#173 (eigene Datei + isolierte Funktion).

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved gesture recognition by preserving the original touch-down point during long swipes, reducing misclassification when earlier samples are no longer retained.
* **Tests**
  * Added new test coverage for gesture anchoring behavior, including empty/edge cases and long swipe scenarios that validate correct gesture classification and direction mapping.

### CODE REVIEW

**Comment by cl445** [comment]
@coderabbitai review

**Comment by cl445** [comment]
@coderabbitai full review

**Comment by cl445** [comment]
@coderabbitai full review

**Comment by cl445** [comment]
@coderabbitai full review

**Comment by cl445** [comment]
@coderabbitai full review

**Comment by cl445** [comment]
@coderabbitai full review

**Comment by cl445** [comment]
@coderabbitai full review

**Comment by cl445** [comment]
@coderabbitai full review

File: wurstfingerTests/OriginAnchoringTests.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        for x in stride(from: 56, through: 4, by: -4) {
+            evicted.append(CGPoint(x: CGFloat(x), y: 0))
+        }
+
+        // Re-anchoring restores (0,0) as the origin, so the gesture reads as a
+        // rightward return swipe again.
+        let anchored = KeyGestureRecognizer.anchoringOrigin(evicted)
+        let features = GestureFeatures.extract(from: anchored)
+        #expect(features.isReturn)
+        #expect(KeyGestureRecognizer.angleToGestureType(features.maxDisplacementAngle) == .swipeRight)
```
[the remark this answers is not in the document]
Comment (cl445) [reply wurstfingerTests/OriginAnchoringTests.swift]: Übernommen: Der Test prüft jetzt über die Produktions-Pipeline `KeyGestureRecognizer.classify(positions:)` statt einer manuellen Feature-Extraktion.

**Comment by cl445** [comment]
@coderabbitai full review
